#!/bin/sh /usr/share/dpatch/dpatch-run
## zone2ldap.dpatch by Matthijs Mohlmann <matthijs@cacholong.nl>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix zone2ldap binary so it works with newer versions of slapd.

@DPATCH@
diff -urNad pdns-2.9.16/pdns/backends/bind/zone2ldap.cc /tmp/dpep.jQiGci/pdns-2.9.16/pdns/backends/bind/zone2ldap.cc
--- pdns-2.9.16/pdns/backends/bind/zone2ldap.cc	2004-12-05 15:28:56.000000000 +0000
+++ /tmp/dpep.jQiGci/pdns-2.9.16/pdns/backends/bind/zone2ldap.cc	2004-12-05 15:29:19.000000000 +0000
@@ -72,8 +72,6 @@
 		g_objects[domain2] = true;
 
 		cout << "changetype: add" << endl;
-		cout << "objectclass: top" << endl;
-		if( domain2 == g_zonename ) { cout << "objectclass: dcobject" << endl; }   // only necessary for phpgeneral
 		cout << "objectclass: dnsdomain2" << endl;
 		cout << "objectclass: domainrelatedobject" << endl;
 		cout << "dc: " << host << endl;
@@ -83,6 +81,7 @@
 	else
 	{
 		cout << "changetype: modify" << endl;
+		cout << "add: " << qtype << "Record" << endl;
 	}
 
 	cout << qtype << "Record: ";
@@ -113,8 +112,7 @@
 
 			cout << "dn: " << dn << g_basedn << endl;
 			cout << "changetype: add" << endl;
-			cout << "objectclass: top" << endl;
-			cout << "objectclass: dcobject" << endl;
+			cout << "objectclass: dnsdomain2" << endl;
 			cout << "objectclass: domainrelatedobject" << endl;
 			cout << "dc: " << parts[i] << endl;
 			cout << "associateddomain: " << net << endl << endl;
@@ -130,8 +128,6 @@
 		g_objects[domain2] = true;
 
 		cout << "changetype: add" << endl;
-		cout << "objectclass: top" << endl;
-		if( domain2 == g_zonename ) { cout << "objectclass: dcobject" << endl; }   // only necessary for phpgeneral
 		cout << "objectclass: dnsdomain2" << endl;
 		cout << "objectclass: domainrelatedobject" << endl;
 		cout << "dc: " << parts[0] << endl;
@@ -141,6 +137,7 @@
 	else
 	{
 		cout << "changetype: modify" << endl;
+		cout << "add: " << qtype << "Record" << endl;
 	}
 
 	cout << qtype << "Record: ";
